modules/adplayerproVideoProvider.js

```diff
@@ -0,0 +1,439 @@
+    // }
+
+    if (!document.getElementById(divId)) {
+      triggerSetupFailure(-3, setupFailMessage + ': No div found with id ' + divId);
+      return;
+    }
+
+    if (!playerConfig || !playerConfig.placementId) {
+      triggerSetupFailure(-4, setupFailMessage + ': placementId is required in playerConfig');
+      return;
+    }
+
+    triggerSetupComplete();
+  }
+
+  function getId() {
+    return divId;
+  }
+
+  function getOrtbVideo() {
+    supportedMediaTypes = supportedMediaTypes || utils.getSupportedMediaTypes(MEDIA_TYPES);
+
+    const video = {
+      mimes: supportedMediaTypes,
+      protocols: [
+        PROTOCOLS.VAST_2_0,
+        PROTOCOLS.VAST_3_0,
+        PROTOCOLS.VAST_4_0,
+        PROTOCOLS.VAST_2_0_WRAPPER,
+        PROTOCOLS.VAST_3_0_WRAPPER,
+        PROTOCOLS.VAST_4_0_WRAPPER
+      ],
+      // h: player.getHeight(),
+      // w: player.getWidth(),
+      placement: utils.getPlacement(playerConfig.advertising),
+      maxextended: -1, // extension is allowed, and there is no time limit imposed.
+      boxingallowed: 1,
+      playbackmethod: [utils.getPlaybackMethod(config)],
+      playbackend: 1,
+      api: [
+        API_FRAMEWORKS.VPAID_2_0,
+        API_FRAMEWORKS.OMID_1_0
+      ],
+    };
+
+    return video;
+  }
+
+  function getOrtbContent() {
+  }
+
+  function setAdTagUrl(adTagUrl, options) {
+    setupPlayer(playerConfig, adTagUrl || options.adXml)
+  }
+
+  function onEvent(externalEventName, callback, basePayload) {
+    if (externalEventName === SETUP_COMPLETE) {
+      setupCompleteCallbacks.push(callback);
+      return;
+    }
+
+    if (externalEventName === SETUP_FAILED) {
+      setupFailedCallbacks.push(callback);
+      return;
+    }
+
+    let getEventPayload;
+
+    switch (externalEventName) {
+      case AD_REQUEST:
+      case AD_PLAY:
+      case AD_PAUSE:
+      case AD_LOADED:
+      case AD_STARTED:
+      case AD_IMPRESSION:
+      case AD_CLICK:
+      case AD_SKIPPED:
+      case AD_ERROR:
+      case AD_COMPLETE:
+      case MUTE:
+      case VOLUME:
+      case ERROR:
+      case PLAYER_RESIZE:
+        getEventPayload = e => ({
+          height: player.getAdHeight(),
+          width: player.getAdWidth(),
+        });
+        break;
+      default:
+        return;
+    }
+
+    // eslint-disable-next-line no-unreachable
+    const playerEventName = utils.getPlayerEvent(externalEventName);
+    const eventHandler = getEventHandler(externalEventName, callback, basePayload, getEventPayload)
+    player && player.on(playerEventName, eventHandler);
+    callbackStorage.storeCallback(playerEventName, eventHandler, callback);
+  }
+
+  function offEvent(event, callback) {
+    const playerEventName = utils.getPlayerEvent(event);
+    const eventHandler = callbackStorage.getCallback(playerEventName, callback);
+    if (eventHandler) {
+      player && player.off(playerEventName);
+    } else {
+      player && player.off(playerEventName, eventHandler);
+    }
```

Comment:
was this meant to be inverted ?

```suggestion
    if (eventHandler) {
      player && player.off(playerEventName, eventHandler);
    } else {
      player && player.off(playerEventName);
    }
```
seems strange to check for eventHandler but only reference it in the else

Reply:
Thank you for your review. Yes, this is a mistake. I fixed it and added a test for this case.